dist/api_data.py

```diff
@@ -218,6 +218,10 @@ def __ge__(self, other):
             the bucket indicating the location for this table'''),
         Config('bucket_prefix', '', r'''
             the unique bucket prefix for this table'''),
+        Config('cache_directory', '', r'''
+            a directory to store locally cached versions of files in the storage source.  By
+            default, it is named with \c "-cache" appended to the bucket name.  A relative
+            directory name is relative to the home directory'''),
```

Comment:
Not directly related to these lines, it is just the top of the review. I think `tiered_abort` should also be modified to remove the prepending of the `working_dir` to the bucket name. The code for `run_workload` no longer needs the buffer argument that constructs the path because the whole thing can just use the `BUCKET` macro since it is all relative.

Reply:
@sueloverso, I looked at `tiered_abort`, and in fact it doesn't concatenate the working dir as far as I can tell.  It reuses `buf` and `bucket`, I modified it to be more understandable (I think) and makes it clearer that it doesn't have that problem.  I'll include those changes in this PR to see if you agree.

BTW, `tiered_abort` (and possibly other utilities) have a different problem related to `cmake`: `testutil_build_dir` assumes that `build_posix` is the build directory where you find things, and the build directory is `build` with `cmake`.  Also `tiered_abort` uses `"/.libs/"` in a pathname, which doesn't work with `cmake`.  I'll open a new ticket for these.

Reply:
To clarify, `tiered_abort` _does_ concatenate the build directory with the storage source library name (which is appropriate), but it doesn't concatenate anything with the bucket name when it hands it to `wiredtiger_open`.

Reply:
I opened WT-8076 to track the cmake issues.